actor/src/main/scala/org/apache/pekko/pattern/RetrySupport.scala

```diff
@@ -46,37 +47,132 @@ trait RetrySupport {
+   *
+   * <b>Example usage:</b>
+   *
+   * {{{
+   * def possiblyFailing(): Future[Something] = ???
+   * val shouldRetry: (Something, Throwable) => throwable ne null
+   * val withRetry: Future[Something] = retry(attempt = possiblyFailing, shouldRetry, attempts = 10)
+   * }}}
+   *
+   * @param attempt     the function to be attempted
+   * @param shouldRetry the predicate to determine if the attempt should be retried
+   * @param attempts    the maximum number of attempts
+   * @param ec          the execution context
+   * @return the result future which maybe retried
+   *
+   * @since 1.1.0
+   */
+  def retry[T](attempt: () => Future[T],
+      shouldRetry: (T, Throwable) => Boolean,
+      attempts: Int)(implicit ec: ExecutionContext): Future[T] = {
+    RetrySupport.retry(attempt, shouldRetry, attempts, ConstantFun.scalaAnyToNone, attempted = 0)(ec, null)
+  }
+
   /**
    * Given a function from Unit to Future, returns an internally retrying Future.
    * The first attempt will be made immediately, each subsequent attempt will be made with a backoff time,
    * if the previous attempt failed.
    *
    * If attempts are exhausted the returned future is simply the result of invoking attempt.
    * Note that the attempt function will be invoked on the given execution context for subsequent
-   * tries and therefore must be thread safe (not touch unsafe mutable state).
+   * tries and therefore must be thread safe (i.e. not touch unsafe mutable state).
+   *
+   * <b>Example usage:</b>
+   *
+   * {{{
+   * protected val sendAndReceive: HttpRequest => Future[HttpResponse]
+   * private val sendReceiveRetry: HttpRequest => Future[HttpResponse] = (req: HttpRequest) => retry[HttpResponse](
+   *   attempt = () => sendAndReceive(req),
+   *   attempts = 10,
+   *   minBackoff = 1.seconds,
+   *   maxBackoff = 2.seconds,
+   *   randomFactor = 0.5
+   * )
+   * }}}
+   *
+   * @param minBackoff   minimum (initial) duration until the child actor will
+   *                     started again, if it is terminated
+   * @param maxBackoff   the exponential back-off is capped to this duration
+   * @param randomFactor after calculation of the exponential back-off an additional
+   *                     random delay based on this factor is added, e.g. `0.2` adds up to `20%` delay.
+   *                     In order to skip this additional delay pass in `0`.
+   */
+  def retry[T](
+      attempt: () => Future[T],
+      attempts: Int,
+      minBackoff: FiniteDuration,
+      maxBackoff: FiniteDuration,
+      randomFactor: Double)(implicit ec: ExecutionContext, scheduler: Scheduler): Future[T] = {
+    require(attempt != null, "Parameter attempt should not be null.")
+    require(minBackoff != null, "Parameter minBackoff should not be null.")
+    require(maxBackoff != null, "Parameter maxBackoff should not be null.")
+    require(minBackoff > Duration.Zero, "Parameter minBackoff must be > 0")
```

Comment:
why we compare Scala FiniteDuration and Java Duration? looks like should be `minBackoff.length > 0` or `minBackoff.toNano` > 0

Reply:
This is a `Scala` Duration. see the imports above,